src/components/structures/TimelinePanel.tsx

```diff
@@ -773,7 +773,7 @@ class TimelinePanel extends React.Component<IProps, IState> {
     private onRoomTimelineReset = (room: Room, timelineSet: EventTimelineSet): void => {
         if (timelineSet !== this.props.timelineSet) return;
 
-        if (this.messagePanel.current && this.messagePanel.current.isAtBottom()) {
+        if (this.canResetTimeline()) {
```

Comment:
The logic is equivalent to before:


https://github.com/matrix-org/matrix-react-sdk/blob/bc60e59eda904fd35a0b7c3f5b6a041114e45034/src/components/structures/TimelinePanel.tsx#L781